pages/markOrderAsPaid.js

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import { graphql } from 'react-apollo';
+import gql from 'graphql-tag';
+
+import { getStripe } from '../lib/stripe';
+import MessageBox from '../components/MessageBox';
+import Container from '../components/Container';
+import { withUser } from '../components/UserProvider';
+import AuthenticatedPage from '../components/AuthenticatedPage';
+
+class MarkOrderAsPaidPage extends React.Component {
+  static getInitialProps({ query }) {
+    return { id: parseInt(query.id) };
+  }
+
+  static propTypes = {
+    /** OrderId */
+    id: PropTypes.number.isRequired,
+    /** @ignore from graphql */
+    markOrderAsPaid: PropTypes.func.isRequired,
+    /** @ignore from withUser */
+    loadingLoggedInUser: PropTypes.bool.isRequired,
+    /** @ignore from withUser */
+    LoggedInUser: PropTypes.object,
+  };
+
+  state = {
+    status: MarkOrderAsPaidPage.SUBMITTING,
+    isRequestSent: false,
+    error: null,
+  };
+
+  componentDidMount() {
+    if (!this.props.loadingLoggedInUser && this.props.LoggedInUser) {
+      return this.triggerRequest();
+    }
+  }
+
+  componentDidUpdate() {
+    if (!this.state.isRequestSent && !this.props.loadingLoggedInUser && this.props.LoggedInUser) {
+      return this.triggerRequest();
+    }
+  }
+
+  static SUBMITTING = 1;
+  static SUCCESS = 2;
+  static ERROR = 3;
+
+  async triggerRequest() {
+    try {
+      this.setState({ isRequestSent: true });
+      const res = await this.props.markOrderAsPaid(this.props.id);
+      const orderConfirmed = res.data.markOrderAsPaid;
+      if (orderConfirmed.stripeError) {
+        this.handleStripeError(orderConfirmed);
+      } else {
+        this.setState({ status: MarkOrderAsPaidPage.SUCCESS });
+      }
+    } catch (e) {
+      const error = get(e, 'graphQLErrors.0') || e;
+      this.setState({ status: MarkOrderAsPaidPage.ERROR, error: error.message });
+    }
+  }
+
+  handleStripeError = async ({ id, stripeError: { message, account, response } }) => {
+    if (!response) {
+      this.setState({ status: MarkOrderAsPaidPage.ERROR, error: message });
+      return;
+    }
+    if (response.paymentIntent) {
+      const stripe = await getStripe(null, account);
+      const result = await stripe.handleCardAction(response.paymentIntent.client_secret);
+      if (result.error) {
+        this.setState({ status: MarkOrderAsPaidPage.ERROR, error: result.error.message });
+      }
+      if (result.paymentIntent && result.paymentIntent.status === 'requires_confirmation') {
+        this.triggerRequest({ id });
+      }
+    }
```

Comment:
I don't understand how Stripe is related here

Reply:
Removed.